hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractSystemUndeleteHandler.java

```diff
@@ -46,18 +46,20 @@ public ContractSystemUndeleteHandler() {
 
     @Override
     public void preHandle(@NonNull final PreHandleContext context) throws PreCheckException {
+        // this will never actually get called
+        // because pureChecks will always throw
         throw new PreCheckException(NOT_SUPPORTED);
     }
 
     @Override
     public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckException {
-        // nothing to do
+        throw new PreCheckException(NOT_SUPPORTED);
```

Comment:
Should we throw in `pureChecks` ? I am not sure why it was reverted in `UncheckedSubmit` https://github.com/hashgraph/hedera-services/pull/12874/files

Reply:
Per discussion, this is the correct behavior.